src/OrleansConsulUtils/SerializableMembershipTypes.cs

```diff
@@ -0,0 +1,130 @@
+        internal ulong LastIndex { get; set; }      //Used to compare CAS value on update
+
+        //Public properties are serialized to the KV.Value
+        [JsonProperty]
+        public String Hostname { get; set; }
+
+        [JsonProperty]
+        public Int32 ProxyPort { get; set; }
+
+        [JsonProperty]
+        public DateTime StartTime { get; set; }
+
+        [JsonProperty]
+        public DateTime IAmAliveTime { get; set; }
+
+        [JsonProperty]
+        public SiloStatus Status { get; set; }
+
+        [JsonProperty]
+        public List<SuspectingSilo> SuspectingSilos { get; set; }
+
+        internal ConsulSiloRegistration()
+        {
+            SuspectingSilos = new List<SuspectingSilo>();
+        }
+
+        internal static String ParseDeploymentKVPrefix(String deploymentId)
+        {
+            return String.Format("{0}{1}{2}", DeploymentKVPrefix, KeySeparator, deploymentId);
+        }
+
+        internal static String ParseDeploymentSiloKey(String deploymentId, SiloAddress siloAddress)
+        {
+            return String.Format("{0}{1}{2}", ParseDeploymentKVPrefix(deploymentId), KeySeparator, siloAddress.ToParsableString());
+        }
+
+        internal static ConsulSiloRegistration FromKVPair(String deploymentId, KVPair kv)
+        {
+            var ret = JsonConvert.DeserializeObject<ConsulSiloRegistration>(Encoding.UTF8.GetString(kv.Value));
+            var keyParts = kv.Key.Split(KeySeparator);
+            ret.Address = SiloAddress.FromParsableString(keyParts.Last());
+            ret.DeploymentId = deploymentId;
+            ret.LastIndex = kv.ModifyIndex;
+            return ret;
+        }
+
+        internal static ConsulSiloRegistration FromMembershipEntry(String deploymentId, MembershipEntry entry, String etag)
+        {
+            var ret = new ConsulSiloRegistration
+            {
+                DeploymentId = deploymentId,
+                Address = entry.SiloAddress,
+                LastIndex = Convert.ToUInt64(etag),
+                Hostname = entry.HostName,
+                ProxyPort = entry.ProxyPort,
+                StartTime = entry.StartTime,
+                IAmAliveTime = entry.IAmAliveTime,
+                Status = entry.Status,
+                SuspectingSilos = entry.SuspectTimes.Select(silo => new SuspectingSilo { Id = silo.Item1.ToParsableString(), Time = silo.Item2 }).ToList()
+            };
+
+            return ret;
+        }
+
+        internal KVPair ToKVPair()
+        {
+            var siloKV = new KVPair(ParseDeploymentSiloKey(DeploymentId, Address));
+            siloKV.ModifyIndex = LastIndex;
+            siloKV.Value = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(this));
+            return siloKV;
+        }
+
+        internal Tuple<MembershipEntry, String> ToMembershipEntry()
+        {
+            var entry = new MembershipEntry
+            {
+                SiloAddress = Address,
```

Comment:
The ConsulSiloRegistration is assembled with the DeploymentId so it will always be set when you come to convert it to a KVPair.